### PR TITLE
Browser: Fix calculation of greatest child width

### PR DESCRIPTION
**Pull request by NoahR02**
The greatest child width in `BlockFormattingContext.cpp` does not account for the `min-width` of a child.

You can reproduce this issue with the following html document:
```html
<!DOCTYPE html>
<html lang="en">

<head>
  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>Document</title>
  <style>
    .debug_here {
      display: inline-block;
      border: 1px solid red;
    }

    .AvatarStack {
      min-width: 26px;
      min-height: 50px;
    }

    .AvatarStack .AvatarStack-body {
      position: absolute;
      width: 24px;
      height: 24px;
      opacity: 0.3;
      background: red;
    }

    .text {
      display: inline-block;
    }
  </style>
</head>

<body>
  <div class="debug_here">
    <div class="AvatarStack">
      <div class="AvatarStack-body"></div>
    </div>
  </div>
  <div class="text"><a>awesomekling</a><a>LibJS:Only use 1 bit for Cell boolean flag</a></div>
</body>

</html>
```

If you dump the layout blocks, then you will see that `debug_here` has a width of 0px. `debug_here` should be the greatest width of it's child.

### CODE REVIEW

**Review by awesomekling — CHANGES_REQUESTED** [review]

File: Userland/Libraries/LibWeb/Layout/BlockFormattingContext.cpp [hunk 1]
```diff
@@ -771,8 +771,14 @@ float BlockFormattingContext::greatest_child_width(Box const& box)
...
-            if (!child.is_absolutely_positioned())
-                max_width = max(max_width, m_state.get(child).border_box_width());
+            if (!child.is_absolutely_positioned()) {
+                auto min_width = child.computed_values().min_width();
+                // Account for the min width of the child.
+                if (min_width.is_length())
+                    max_width = max(max_width, max(min_width.length().raw_value(), m_state.get(child).border_box_width()));
+                else
+                    max_width = max(max_width, m_state.get(child).border_box_width());
+            }
```
[the remark this answers is not in the document]
Comment (NoahR02) [reply Userland/Libraries/LibWeb/Layout/BlockFormattingContext.cpp:781]: That makes sense. I'll go back soon and look through `layout_block_level_box` to see if I can find where `min-width` is not accounted for.

**Comment by NoahR02** [comment]
I thought this bugfix fixed the problem, but I had some old git changes that messed up the output. Maybe this commit should be moved to another pull request?

This pull request is still in progress. Please disregard the review for now.

**Comment by NoahR02** [comment]
In `InlineFormattingContext.cpp`, if the width is auto then we calculate the shrink fit which in turn calls greatest_child_width. The available space we provide in this scenario is under a intrinsic sizing constraint, so compute_width never has the opportunity to calculate the width correctly because it does not account for min or max width which in turn messes up the calculation of the calling `InlineFormattingContext`.

We cannot accurately calculate the shrink fit  correctly if we return early from `BlockFormattingContext::compute_width`. 
If we remove the if statement then the shrink fit calculation is now correct and produces the correct width:

Test Case:
```html
<!DOCTYPE html><html><head><style>    
    .debug_here {    
      display: inline-block;    
      border: 1px solid red;   
    }    
    .AvatarStack {    
      min-width: 26px;    
      min-height: 50px;    
    }    
</style></head><body><div class="debug_here"><div class="AvatarStack">
```

File: Userland/Libraries/LibWeb/Layout/BlockFormattingContext.cpp [hunk 2]
```diff
@@ -107,9 +107,6 @@ void BlockFormattingContext::compute_width(Box const& box, AvailableSpace const&
 
     auto const& computed_values = box.computed_values();
 
-    if (should_treat_width_as_auto(box, available_space) && available_space.width.is_intrinsic_sizing_constraint())
-        return;
```
Comment (awesomekling) [inline Userland/Libraries/LibWeb/Layout/BlockFormattingContext.cpp:111]: Unfortunately it's not this simple. I'm not exactly sure how this needs to work, but here's an example of something that will break with this change:

```html
<!DOCTYPE html><html><head><style>
  .outer {
    display: inline-block;
  }
  .inner {
    background: orange;
    min-width: 32px;
  }
</style></head><body><div class="outer"><div class="inner">well hello friends</div></div>
```

**Review by awesomekling — CHANGES_REQUESTED** [review]

**Pull request closed without merging**